torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -125,13 +125,15 @@ PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   return array.release();
 }
 
-at::Tensor tensor_from_numpy(PyObject* obj) {
+at::Tensor tensor_from_numpy(PyObject* obj, bool warn_if_not_writeable/*=false*/) {
   if (!PyArray_Check(obj)) {
     throw TypeError("expected np.ndarray (got %s)", Py_TYPE(obj)->tp_name);
   }
   auto array = (PyArrayObject*)obj;
 
-  if (!PyArray_ISWRITEABLE(array)) {
+  // warn_if_not_writable is true when a copy of numpy variable is created.
+  // the warning is suppressed when a copy is being created.
+  if (!PyArray_ISWRITEABLE(array) && !warn_if_not_writeable) {
```

Comment:
`!warn_if_not_writeable` -> `warn_if_not_writeable `

Reply:
Updated in https://github.com/pytorch/pytorch/pull/47271/commits/c9a7efa43c9dc406335e0b6beda757228f9599a9